pkg/utils/helper.go

```diff
@@ -52,6 +54,21 @@ func IsAzureStackCloud() bool {
 	return strings.EqualFold(cloud, AzureStackCloudName)
 }
 
+// IsRunningInAks returns true if the application is running on AKS
+func IsRunningInAks() bool {
+	//TODO refactor the conditional logic this check guards into new Cluster-Type specific components behind an interface
+	//test the non-AKS kubeconfig location, if we find something then this is *not* an AKS cluster
+	_, err := RunCommandOnHost("ls", "/etc/kubernetes/kubelet.conf")
```

Comment:
💡Why is this used to identify if the cluster is `AKS` or not?

Reply:
Yeah,  This could be done more like fall back checks which will help in better, failure error once all fails for `non-aks` scenario.

I would highly recommend this:

1. check for cluster-info : using `kubectl cluster-info` 
    *  For AKS it will be `-aks-fqdn.azmk8s.io.` , and other providers have similar patterns.
2. Also, If we run with in the cluster then we could have used :https://kubernetes.io/docs/tasks/administer-cluster/namespaces/#deleting-a-namespace 
3. check third fallback and then -> graciously fail with right messages.

It is key that we right these functions correctly so the we can identify what all kind cluster we could be extending this support.

cc: @safeermohammed , @arnaud-tincelin or @palma21 - for more ideas to piggy back on this.